projects/core/src/lazy-loading/unified-injector.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  AbstractType,
+  Injectable,
+  InjectFlags,
+  InjectionToken,
+  Injector,
+  Type,
+} from '@angular/core';
+import { LazyModulesService } from './lazy-modules.service';
+import { asapScheduler, Observable } from 'rxjs';
+import { auditTime, filter, map, scan, startWith } from 'rxjs/operators';
+
+const NOT_FOUND_SYMBOL: any = {};
+
+/**
+ * UnifiedInjector provides a way to get instances of tokens not only once, from the root injector,
+ * but also from lazy loaded module injectors that can be initialized over time.
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class UnifiedInjector {
+  /**
+   * Gather all the injectors, with the root injector as a first one
+   *
+   * @private
+   */
+  private injectors$: Observable<Injector> = this.lazyModules.modules$.pipe(
+    map((moduleRef) => moduleRef.injector),
+    startWith(this.rootInjector)
+  );
+
+  constructor(
+    protected rootInjector: Injector,
+    protected lazyModules: LazyModulesService
+  ) {}
+
+  /**
+   * Gen instances for specified tokens.
+   *
+   * When notFoundValue is provided, it will consistently emit once per injector,
+   * even if injector doesn't contain instances for specified token.
+   * Otherwise, emissions will only involve cases, where new instances will be found.
+   *
+   * @param token
+   * @param notFoundValue
+   */
+  get<T>(
+    token: Type<T> | InjectionToken<T> | AbstractType<T>,
+    notFoundValue?: T
+  ): Observable<T> {
+    return this.injectors$.pipe(
+      map((injector, index) =>
+        injector.get<T>(
+          token,
+          notFoundValue ?? NOT_FOUND_SYMBOL,
+          index ? InjectFlags.Self : undefined
```

Comment:
The first inector will always be the `rootInjector`. 

For  all injectors except the root want, we want to use InjectFlag..Self, to not get values from root injector again (as all lazy loaded injectors have rootInjector as a parrent in most cases)

Reply:
1. For `injectors$` the values emitted in total === the values emitted since subscription (because of publishReplay, now it get it). But it's not obvious in this place.

2. I'm just trying to simplify. Can't we get from the `rootInjector` using flag `Self` too?

3. Anyway, a line of code comment explaining the context may be beneficial for future.

Reply:
1. It's not because of publishReplay in source observable, but because of `startWith` in `injectors$` which we are subscribing here directly. .

Reply:
2. I'm afraid, we shouldn't ;-) Even if it could in some default cases.

Reply:
3. :+1:

Reply:
As discussed offline, the first emitted injector is the root injector. And if a token is not found in the root injector, we'd like to lookup providers higher in the injectors hierarchy - in the platform injector or in the SSR engine injector (if run in SSR mode).